pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ConsumerEventListener.java

```diff
@@ -49,4 +50,17 @@
      */
     void becameInactive(Consumer<?> consumer, int partitionId);
 
+
+    /**
+     * Notified when the consumer key_shared_rule is changed.
+     *
+     * @param consumer
+     *            the consumer that originated the event
+     * @param keyPredicate
+     *            Determine whether the key will be consumed by {@param consumer}
+     */
+    default void keySharedRuleChanged(Consumer<?> consumer, Predicate<String> keyPredicate) {
```

Comment:
`keyHashRangesChanged` may be more reasonable here.

Sorry, I'm a little confused here, why do we need a `keyPredicate` here? We should send the latest key hash ranges to the consumer.

Reply:
The `keyPredicate` will be useful in this case : [At #6555, User want know `when a key will be consumed by current consumer` ](https://github.com/apache/pulsar/issues/6555#issuecomment-735347905). 

ah ah ha!.  Sorry. It looks like "I misunderstood the need". 

My thought was: 

In `Mode-failover` message sharding by patition, `ConsumerEventListener` tell user "witch message will be consumed by current consumer".

In `Mode-Key_Shared` message sharding by key, `ConsumerEventListener` tell user "witch message will be consumed by current consumer" too.

How am I supposed to understand "latest key hash range" ?